project-base/infrastructure/google-cloud/gke-cluster.tf

```diff
@@ -12,9 +12,10 @@ resource "google_container_cluster" "primary" {
   node_config {
     oauth_scopes = [
       "https://www.googleapis.com/auth/compute",
-      "https://www.googleapis.com/auth/devstorage.read_only",
+      "https://www.googleapis.com/auth/devstorage.full_control",
       "https://www.googleapis.com/auth/logging.write",
       "https://www.googleapis.com/auth/monitoring",
+      "https://www.googleapis.com/auth/cloud-platform",
```

Comment:
So this is permitting the application to access the bucket without `service-account.json`? If so, maybe you can remove the `kubernetes_secret.gcs-service-account`, `google_service_account_key.gcs-service-account`, `google_service_account.gcs-service-account` and the output variable `gcs-service-account-json-key` from Terraform, as the secret is not used anywhere. I think it was prepared to be used to access the GCS. If not, I might have misunderstood the way deployed app is authorized.